### DISCUSSION TITLE
Setting native version numbers with capacitor

### DISCUSSION THREAD

**Discussion by danielehrhardt** [ionic-team/capacitor#3164] [General] [answered]
With cordova, we could configure a few version numbers using attributes on the tag in config.xml.
How do we set those such that they make it over to the native project when capacitor is synced?
I think this is important for CLI Builds.

**Comment by tavosansal** [comment]
This is currently not supported with capacitor. You have to do it yourself in the native IDEs.

**Comment by jcesarmobile** [comment, marked as answer]
As tavosansal said, you have to do it in the native IDEs.

This is the information about iOS:
https://capacitorjs.com/docs/ios/configuration#configuring-info-plist

If you open the `Info.plist` file as Property List (default), they are the `Bundle version string (short)` field and `Bundle version`. If opening as source code they are `CFBundleShortVersionString` and `CFBundleVersion`. First one is the version string and the second the version number.

For Android you have to change it in the App's `build.gradle` file changing `versionName` and `versionCode`.

If you build in a CI/CD platform, tools such as fastlane allow to increment the version automatically.

**Reply by DEberhardBrelag** [reply]
The part with the `build.gradle` should at least be updated in the capacitors android>configuration page. It has somewhat missleading Infos there since it says we only have to change it in `AndroidManifest.xml` and `strings.xml`, but that doesn't affect the native android build. So mention the `gradle.build` there too might be a good idea.
https://capacitorjs.com/docs/android/configuration

**Reply by jcesarmobile** [reply]
Yeah, there is an issue for fixing it
https://github.com/ionic-team/capacitor-site/issues/60

**Comment by mlynch** [comment]
There's now support for this in Trapeze: https://trapeze.dev

**Comment by justjoshin83** [comment]
I must say, one of the best reasons to use Ionic is to stay as far away from the native IDE's as possible; it's in fact the entire reason I chose this platform in the first place.  Because like me, many of us *don't have macs* so we can't possibly open XCode to do "x,y, and z".    Removing the ability to set build numbers at an "ionic" level in favor of having to do it at the native IDE level seems like a regression for the framework :(  I really hope this doesn't become a trend.

**Comment by vesper8** [comment]
Found this useful package that helps with setting these versions https://github.com/talhahasanzia/x-version
